packages/vats/test/test-lib-board.js

```diff
@@ -49,6 +49,14 @@ test('makeBoard', async t => {
   t.is(idObj2b, 'tooboard012');
 });
 
+test('board values must be scalar keys', async t => {
+  const board = makeBoard();
+  const nonKey = harden({ a: 1 });
```

Comment:
Seems like the patterns API is reasonably well tested elsewhere.

Reply:
Hmm, so what exactly is this testing? The "invalid key type" error seems to come from [collectionManager.js](https://github.com/Agoric/agoric-sdk/blob/efd964db27052459bd55193b5ff82b9f2b8aab1d/packages/swingset-liveslots/src/collectionManager.js#L208), which means that the bad input was _accepted_ by the method guard, which suggests to me that `ValShape` in packages/vats/src/lib-board.js should be `M.scalar()` rather than `M.key()`.

Reply:
Currently, non-scalar keys are not yet implemented by `*MapStore` and `*SetStore`. We make the "Scalar" qualifier explicit in the maker names to emphasize that these implement only a subset of the full MapStore and SetStore semantics.

Non-scalar keys are implemented by CopyMap, CopySet, and CopyBag.

So whether the correct restriction to express is `M.key()` or `M.scalar()` depends on if you're trying to detect a violation of the specified semantics, or a violation of the current implementation restrictions. Either way, please put a comment at the relevant place in the code to explain.

Reply:
> Currently, non-scalar keys are not yet implemented by `*MapStore` and `*SetStore`.

Hmm, then it seems like a problem for e.g. `makeScalarBigMapStore("dummy", { keyShape: M.key() }).init(harden({ label: "not a scalar" }), "val")` to succeed. But more importantly, doesn't that restriction make Board durability (and thus upgradability) an order of magnitude more challenging? How should it store the value-to-id data if not in a durable map?

> We make the "Scalar" qualifier explicit in the maker names to emphasize that these implement only a subset of the full MapStore and SetStore semantics.

D'oh! I should have picked up on/remembered that!

Reply:
`makeScalarBigMapStore("dummy", { keyShape: M.key() }).init(harden({ label: "not a scalar" }), "val")` succeeds? Really? That is very surprising. Are you sure?

Reply:
```sh
$ node --input-type=module -e '
  import "@agoric/swingset-vat/tools/prepare-test-env.js";
  import { M } from "@agoric/store";
  import { makeScalarBigMapStore } from "@agoric/vat-data";
  const bigMap = makeScalarBigMapStore("dummy", { keyShape: M.key() });
  const key = harden({ label: "not a scalar" });
  bigMap.init(key, "val");
  console.log(bigMap.get(key));
'
val
```

Reply:
@FUDCo @warner , do you have any idea how this could possibly be working? Are our Big*Stores no longer limited to scalar keys? Did this happen in switching to sqlite?

Reply:
I don't think we're that lucky; more likely the in-memory implementation set up for testing is just not very restrictive and the `keyShape` argument for `makeScalarBigMapStore` et al. is treated as a _replacement_ for the `M.scalar()` default rather than a _refinement_ of it.